service-food/src/main/kotlin/de/hpi/cloud/food/FoodService.kt

```diff
@@ -68,34 +68,29 @@ class FoodServiceImpl(private val bucket: Bucket) : FoodServiceGrpc.FoodServiceI
         val restaurantId = req.restaurantId?.trim()?.takeIf { it.isNotEmpty() }
         val date = if (req.hasDate()) req.date else null
 
-        val menuItems =
-            if (restaurantId == null && date == null)
-                bucket.query(ViewQuery.from(DESIGN_MENU_ITEM, VIEW_BY_ID)).allRows()
-                    .map { it.document().content() }
-            else {
-                val statement = Select.select("*")
-                    .from(bucket.name())
-                    .where(
-                        and(
-                            x(KEY_TYPE).eq(s("menuItem")),
-                            restaurantId?.let { n(KEY_VALUE, "restaurantId").eq(s(restaurantId)) },
-                            date?.let {
-                                millisToStr(
-                                    n(KEY_VALUE, "date", "millis"),
-                                    "1111-11-11"
-                                ).eq(s(it.toQueryString()))
-                            }
-                        )
-                    )
-                    .orderBy(*descTimestamp(n(KEY_VALUE, "publishedAt")))
-                bucket.query(N1qlQuery.simple(statement, N1qlParams.build().adhoc(false))).allRows()
-                    .map { it.value().getObject(bucket.name()) }
-            }
-
-
-        ListMenuItemsResponse.newBuilder()
-            .addAllItems(menuItems.map { it.parseMenuItem() })
-            .build()
+        val (menuItems, newToken) = paginate(bucket, {
+            where(
+                and(
+                    x(KEY_TYPE).eq(s("menuItem")),
+                    restaurantId?.let { n(KEY_VALUE, "restaurantId").eq(s(restaurantId)) },
+                    date?.let {
+                        millisToStr(
+                            n(KEY_VALUE, "date", "millis"),
+                            "1111-11-11"
```

Comment:
Kein Plan, was die Query machen soll...
Wofür steht "1111-11-11"?

Reply:
Again, this was already here before this PR and is documented in the Couchbase documentation: <https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/datefun.html#fn-date-millis-to-str>